axelrod/tests/strategies/test_darwin.py

```diff
@@ -15,29 +15,45 @@ class TestDarwin(TestPlayer):
         'stochastic': False,
         'makes_use_of': set(),
         'long_run_time': False,
-        'inspects_source': False,
+        'inspects_source': True,
         'manipulates_source': False,
         'manipulates_state': True
     }
 
+    def setUp(self):
+        """Each test starts with a fresh genome."""
+        self.player.genome = [C]
+        super(TestDarwin, self).setUp()
+
+    def test_setup(self):
+        player = self.player()
+        self.assertEqual(player.genome, [C])
+        self.assertEqual(player.history, [])
+
     def test_strategy(self):
         p1 = self.player()
-        p2 = axelrod.Cooperator()
-        self.assertEqual(p1.strategy(p2), C)  # Always cooperate first.
-        for i in range(10):
-            p1.play(p2)
-        self.assertEqual(p1.strategy(p2), C)
+        p1.reset()
+        self.versus_test(axelrod.Cooperator(), [(C, C)] * 5)
+        self.assertEqual(p1.genome, [C] * 5)
```

Comment:
You don't need this extra line, it can be done in the `versus_test`:

From http://axelrod.readthedocs.io/en/latest/tutorials/contributing/strategy/writing_test_for_the_new_strategy.html:

> The function versus_test also accepts a dictionary parameter of attributes to check at the end of the match. For example this test checks if the player’s internal variable opponent_class is set to "Cooperative":

This would look like:

```
self.versus_test(axelrod.Cooperator(), expected_actions=[(C, C)] * 5, attrs={"genome": [5] * 5})
```

I've got to go to class now so will review the rest later but if you catch things like this that'd be great :)